src/AWActionParser.js

```diff
@@ -237,16 +249,41 @@ function resolveCommand(commandName, commandArguments) {
     return command;
 }
 
-function resolveIncompleteCoordinates(coordinates) {
+function resolveIncompleteCoordinates(coordinates, isScale) {
+  const clamp = (num, min, max) => Math.min(Math.max(num, min), max);
+  let [minScale, maxScale] = [0.1, 10];
+
+  let [x,y,z] = [coordinates[0], coordinates[1], coordinates[2]];
+
+  if (isScale) {
+    if (coordinates[0] <= 0)
+      coordinates[0] = 1;
+    if (coordinates[1] <= 0)
+      coordinates[1] = 1;
+    if (coordinates[2] <= 0)
+      coordinates[2] = 1;
+
+    [x,y,z] = [coordinates[0] < 0.1 ? 0.1 : coordinates[0], coordinates[1] < 0.1 ? 0.1 : coordinates[1], coordinates[2] < 0.1 ? 0.1 : coordinates[2]];
+    if (coordinates.length == 1) {
+      return {x: x, y: x, z: x};
```

Comment:
For this kind of dict declaration, you can also just do something like `{x, y, z}`, which will take the name of the variables as names for the keys :)

You can apply this everywhere else in the code as well.